src/array.jl

```diff
@@ -21,15 +21,15 @@ function reftype(sz::Int)
 end
 
 """
-    CategoricalArray{T}(uninitialized, dims::Dims; ordered::Bool=false)
-    CategoricalArray{T}(uninitialized, dims::Int...; ordered::Bool=false)
+    CategoricalArray{T}(undef, dims::Dims; ordered::Bool=false)
+    CategoricalArray{T}(undef, dims::Int...; ordered::Bool=false)
 
-Construct an uninitialized `CategoricalArray` with levels of type `T` and dimensions `dim`.
+Construct an undef `CategoricalArray` with levels of type `T` and dimensions `dim`.
```

Comment:
I think it's clearer to still call it "uninitialized" in the doc string, even though the implementation is in terms of `undef`.

Reply:
Totally understand. There are a couple of other occurrences in doc strings here though, you might want to readjust those as well.